lib/src/arg_parser.dart

```diff
@@ -210,7 +212,9 @@ class ArgParser {
   /// be included in [usage].
   ///
   /// The [allowedHelp] argument is a map from values in [allowed] to
-  /// documentation for those values that will be included in [usage].
+  /// documentation for those values that will be included in [usage]. If this
+  /// option is provided and [allowed] is `null` the allowed options will be the
+  /// map keys.
```

Comment:
Does this mean there's no way to show help options for some allowed values while still allowing an open-ended set? That seems like an unfortunate restriction. For example, test runner (which doesn't actually use args, but just as an example) only shows a subset of the most common values for some options but allows a larger set.

Reply:
> Does this mean there's no way to show help options for some allowed values while still allowing an open-ended set?

Hmm, yeah it does. That wasn't a tested scenario.

Maybe we should just punt on this for now. It does still seem like a usability bug to need to provide both arguments to get the typical desired behavior, but I'm also not eager to introduce a third argument to support the use case of specific option docs + open ended input.

Reply:
I updated this PR to revert my change and add a test for this behavior. I likely won't bother to pursue this further than that.

PTAL.

Reply:
LGTM!